lib/std/channels.nim

```diff
@@ -477,25 +465,25 @@ template trySend*[T](c: Channel[T], src: T): bool =
 
 func tryRecv*[T](c: Channel[T], dst: var T): bool {.inline.} =
   ## Receives item from the channel(non blocking).
-  channelReceive(c, dst.addr, sizeof(dst), true)
+  recvMpmc(c.d, dst.addr, sizeof(dst), true)
 
 func send*[T](c: Channel[T], src: sink Isolated[T]) {.inline.} =
   ## Sends item to the channel(blocking).
   var data = src.extract
-  discard channelSend(c, data, sizeof(data), false)
+  discard sendMpmc(c.d, data.unsafeAddr, sizeof(data), false)
   wasMoved(data)
 
 template send*[T](c: var Channel[T]; src: T) =
   ## Helper templates for `send`.
   send(c, isolate(src))
 
-func recv*[T](c: Channel[T], dst: var T) {.inline.} =
+func recv*[T](c: Channel[T]): T {.inline.} =
   ## Receives item from the channel(blocking).
-  discard channelReceive(c, dst.addr, sizeof(dst), false)
+  discard recvMpmc(c.d, result.addr, sizeof(result), false)
 
 func recvIso*[T](c: Channel[T]): Isolated[T] {.inline.} =
   var dst: T
-  discard channelReceive(c, dst.addr, sizeof(dst), false)
+  discard recvMpmc(c.d, dst.addr, sizeof(dst), false)
```

Comment:
isn't `recvIso` redundant now, given that we have the more convenient unboxed `func recv*[T](c: Channel[T]): T {.inline.} =` ?

Reply:
fixed